core/src/test/java/com/microsoft/applicationinsights/internal/channel/common/TelemetryBufferTest.java

```diff
@@ -347,6 +357,89 @@ public void testFlushWithSevenInTheBuffer() throws Exception {
         testFlushWithData(7);
     }
 
+    @Test
+    public void testSetTransmitBufferTimeoutInSecondsShorterTime() {
+        MockSender mockSender = new MockSender()
+                .setExpectedNumberOfScheduleSendCalls(0)
+                .setExpectedNumberOfSendNowCalls(1)
+                .setExpectedTelemetriesNumberInScheduleSend(0)
+                .setExpectedTelemetriesNumberInSendNow(2);
+
+        // Create a buffer with max buffer size of 10 and timeout of 10 seconds
+        TelemetryBuffer testedBuffer = new TelemetryBuffer(mockSender, 10, 30);
+
+        for (int i = 0; i < 2; ++i) {
+            testedBuffer.add("mockTelemetry");
+        }
+        testedBuffer.setTransmitBufferTimeoutInSeconds(1);
+
+        mockSender.waitForFinish(1L);
+    }
+
+    @Test
+    public void testSetMaxTelemetriesInBatchWithSmallerSize() {
+        MockSender mockSender = new MockSender()
+                .setExpectedNumberOfScheduleSendCalls(0)
+                .setExpectedNumberOfSendNowCalls(1)
+                .setExpectedTelemetriesNumberInScheduleSend(0)
+                .setExpectedTelemetriesNumberInSendNow(2);
```

Comment:
Please use a constant for the telemetries in SendNow instead of just 2 (here, and below)